File: urbanpulse-backend/src/main/java/com/urbanpulse/service/WorkerService.java
package com.urbanpulse.service;

import com.urbanpulse.entity.Worker;
import java.util.List;

public interface WorkerService {

    Worker saveWorker(Worker worker);

    List<Worker> getAllWorkers();
    Worker updateWorker(Long id, Worker worker);
    void deleteWorker(Long id);

}

File: urbanpulse-frontend/src/services/workerService.js
import axiosInstance from "../api/axiosConfig";

const workerService = {

  getAllWorkers: async () => {
    const response = await axiosInstance.get("/workers");
    return response.data;
  },

  createWorker: async (worker) => {
    const response = await axiosInstance.post(
      "/workers",
      worker
    );
    return response.data;
  },

  updateWorker: async (id, worker) => {
    const response = await axiosInstance.put(
      `/workers/${id}`,
      worker
    );
    return response.data;
  },

  deleteWorker: async (id) => {
    const response = await axiosInstance.delete(
      `/workers/${id}`
    );
    return response.data;
  }

};

export default workerService;